python/veles/server/conn.py

```diff
@@ -66,6 +77,230 @@ def get_node_norefresh(self, obj_id):
             self.objs[obj_id] = res
             return res
 
+    def _check_ok_gone(self, el):
+        node = self.get_node_norefresh(el.node)
+        return node.node is None
+
+    def _check_ok_parent(self, el):
+        node = self.get_node_norefresh(el.node)
+        if node.node is None:
+            return False
+        return node.node.parent == el.parent
+
+    def _check_ok_pos(self, el):
+        node = self.get_node_norefresh(el.node)
+        if node.node is None:
+            return False
+        return (
+            node.node.pos_start == el.pos_start and
+            node.node.pos_end == el.pos_end)
+
+    def _check_ok_tags(self, el):
+        node = self.get_node_norefresh(el.node)
+        if node.node is None:
+            return False
+        return node.node.tags == el.tags
+
+    def _check_ok_tag(self, el):
+        node = self.get_node_norefresh(el.node)
+        if node.node is None:
+            return False
+        return (el.tag in node.node.tags) == el.present
+
+    def _check_ok_attr(self, el):
+        node = self.get_node_norefresh(el.node)
+        if node.node is None:
+            return False
+        return node.node.attr.get(el.key) == el.data
+
+    def _check_ok_data(self, el):
+        node = self.get_node_norefresh(el.node)
+        if node.node is None:
+            return False
+        return self.db.get_data(node.id, el.key) == el.data
+
+    def _check_ok_bindata_size(self, el):
+        node = self.get_node_norefresh(el.node)
+        if node.node is None:
+            return False
+        return node.node.bindata.get(el.key, 0) == el.size
+
+    def _check_ok_bindata(self, el):
+        node = self.get_node_norefresh(el.node)
+        if node.node is None:
+            return False
+        data = self.db.get_bindata(node.id, el.key, el.start, el.end)
+        return data == el.data
+
+    def _check_ok_list(self, el):
+        parent = self.get_node_norefresh(el.parent)
+        if parent.node is None and parent != self.root:
+            return False
+        nodes = self.db.list(el.parent, el.tags, el.pos_filter)
+        return el.nodes == nodes
+
+    def _checks_ok(self, checks):
+        for ch in checks:
+            handlers = {
```

Comment:
nit. it probably won't have any noticeable effect on performance, but it hurt me when I saw this dict constructed inside loop :P